### ISSUE TITLE
Incoming call ring is not linked to proper volume channel

### ISSUE THREAD

**Issue by serrq** [session-foundation/session-android#222]
I noticed that incoming call ringing is linked wrongly to "media" volume instead of "calls".

Thus if I have apps volume low I don't receive any sound from my speaker.

Please fix it.

**Comment by GregHabiskis** [comment]
I'm on v1.27.0 Android and I can confirm this issue still exists.

**Comment by ThomasSession** [comment]
Thanks for raising this. We are tracking this issue internally with ticket SES-4529.

**Comment by ThomasSession** [comment]
This is fixed on a PR which will come out with 1.28.2 in the coming weeks

**Comment by GregHabiskis** [comment]
> This is fixed on a PR which will come out with 1.28.2 in the coming weeks
>

I updated to v1.28.2. It's still not fixed yet.

**Comment by ThomasSession** [comment]
> > This is fixed on a PR which will come out with 1.28.2 in the coming weeks
> 
> I updated to v1.28.2. It's still not fixed yet.

Yeah the version I mentioned above has become 1.29.0 because we added a lot of other changes. 
The 1.28.2 is a different one with some quick fixes to some issues we found in 1.28.1.
Sorry for the confusion.
1.29.0 is on its last staging of testing so hopefully it will be coming out soon.

**Comment by GregHabiskis** [comment]
>Yeah the version I mentioned above has become 1.29.0 because we added a lot of other changes.
>
I updated to v1.29.0. I can confirm that the volume channel issue finally has been fixed.
But now, one of my phones is not receiving voice calls. The one that is working completely fine is Realme C25s RMX3195 on Android 13, and the one not receiving voice calls is Tecno Spark 7 KF6k on Android 11. I confirmed that both are set to the same privacy, notifications, & conversations settings and on the v1.29.0 v8a release.

**Comment by ThomasSession** [comment]
> > Yeah the version I mentioned above has become 1.29.0 because we added a lot of other changes.
> 
> I updated to v1.29.0. I can confirm that the volume channel issue finally has been fixed. But now, one of my phones is not receiving voice calls. The one that is working completely fine is Realme C25s RMX3195 on Android 13, and the one not receiving voice calls is Tecno Spark 7 KF6k on Android 11. I confirmed that both are set to the same privacy, notifications, & conversations settings and on the v1.29.0 v8a release.

We'll look into it. I just tried on a few devices here and the calls work fine. 
If one of your device works could there be anything wrong with the other device? You mentioned all the settings being fine, so I'm not exactly sure what it could be.
Maybe export the logs to see if there are exceptions.

**Comment by GregHabiskis** [comment]
>Maybe export the logs
>
I've emailed the logs. Kindly check them.

**Comment by ThomasSession** [comment]
> > Maybe export the logs
> 
> I've emailed the logs. Kindly check them.

I took a look at the logs and it's unclear which side this was on, but it seems that phone tried to make calls and the recipient was marked as unavailable.
It also seems some call notification might have been missed due to battery optimisation settings or "Do not disturb" mode being on.

**Comment by GregHabiskis** [comment]
>the recipient was marked as unavailable.
>
Both phones were under same wifi network

>some call notification might have been missed due to battery optimisation settings or "Do not disturb" mode being on.
>
I disabled both of them beforehand.

**Comment by ThomasSession** [comment]
> > the recipient was marked as unavailable.
> 
> Both phones were under same wifi network
> 
> > some call notification might have been missed due to battery optimisation settings or "Do not disturb" mode being on.
> 
> I disabled both of them beforehand.

Nothing obvious was shown on the logs. Did you send the logs from the device that doesn't work? The one not receiving the call?
I will close this ticket as this is about the volume channel, which is now fixed, but feel free to open a new ticket if that call issue persists.

**Comment by GregHabiskis** [comment]
>Did you send the logs from the device that doesn't work? The one not receiving the call?
>
Yes. I tested this on Realme C25s Android 13, Realme 7 pro Android 12, and Tecno Spark 7 KF6k Android 11. I'm facing this issue only on Tecno and the other two is working fine.
